File: workspaces-good/packages/division/src/capability_division_analyzer.ts
import { CalculatorProtocol } from "calculator-shared/src/contract_calculator_protocol";
import { ExpressionVO } from "calculator-shared/src/taxonomy_expression_vo";
import { ResultVO, createResult } from "calculator-shared/src/taxonomy_result_vo";

export class DivisionAnalyzer implements CalculatorProtocol {
  evaluate(expr: ExpressionVO): ResultVO | null {
    if (expr.right === 0) return null;
    const value = expr.left / expr.right;
    return createResult(expr.left, "/", expr.right, value);
  }
}

File: workspaces-good/modules/division/src/capability_division_analyzer.py
from typing import Optional

from shared.src.contract_calculator_protocol import CalculatorProtocol
from shared.src.taxonomy_expression_vo import ExpressionVO
from shared.src.taxonomy_result_vo import ResultVO, create_result


class DivisionAnalyzer(CalculatorProtocol):
    def evaluate(self, expr: ExpressionVO) -> Optional[ResultVO]:
        if expr.right == 0:
            return None
        return create_result(expr.left, expr.op, expr.right, expr.left / expr.right)
